examples/blake2f.rs

```diff
@@ -0,0 +1,1377 @@
+use chiquito::{
+    frontend::dsl::{
+        cb::{eq, select, table},
+        lb::LookupTable,
+        super_circuit, CircuitContext, StepTypeSetupContext, StepTypeWGHandler,
+    },
+    plonkish::{
+        backend::halo2::{chiquitoSuperCircuit2Halo2, ChiquitoHalo2SuperCircuit},
+        compiler::{
+            cell_manager::{MaxWidthCellManager, SingleRowCellManager},
+            config,
+            step_selector::SimpleStepSelectorBuilder,
+        },
+        ir::sc::SuperCircuit,
+    },
+    poly::ToExpr,
+    sbpir::query::Queriable,
+};
+use halo2_proofs::{
+    dev::MockProver,
+    halo2curves::{bn256::Fr, group::ff::PrimeField},
+};
+use std::hash::Hash;
+
+pub const IV_LENGTH: usize = 8;
+pub const SIGMA_VECTOR_LENGTH: usize = 16;
+pub const R1: usize = 32;
```

Comment:
Perhaps we are using usize here where we should use u32 or u64. usize is for indexing arrays, their lenght or ranges. But not for any integer.

https://www.reddit.com/r/rust/comments/4lql5l/beginner_question_should_i_use_usize_and_isize/